edx_sysadmin/utils/utility.py

```diff
@@ -21,4 +40,230 @@ def get_course_by_id(course_key, depth=0):
     if course:
         return course
     else:
-        raise Http404(u"Course not found: {}.".format(six.text_type(course_key)))
+        raise Http404(f"{_('Course not found')}: {course_key}.")
+
+
+def get_registration_required_extra_fields():
+    """
+    It processes and returns a list of extra fields which are required for User account
+    registration through Registration API "/user_api/v1/account/registration/", depending
+    upon "settings.REGISTRATION_EXTRA_FIELDS" environment setting
+
+    Arguments:
+    None
+
+    Returns:
+    extra_fields (list) - list of required fields from "settings.REGISTRATION_EXTRA_FIELDS"
+    """
+    extra_fields = []
+    for field, status in settings.REGISTRATION_EXTRA_FIELDS.items():
+        if status == "required":
+            extra_fields.append(field)
+    return extra_fields
+
+
+def get_registration_required_extra_fields_with_values():
+    """
+    It maps registration required extra fields with some pre-defined default values to create
+    django form fields dynamically
+
+    Arguments:
+    None
+
+    Returns:
+    extra_fields (dict) - contains all required fields and their mapping with default values
+    and form type.
+    {
+        ...
+        "FIELD_NAME": {
+            "field_type": "FIELD_TYPE",
+            "default_value": "DEFAULT_VALUE",     # if applicable
+            "choices": "LIST_OF_OPTIONS"          # if applicable
+        }
+        ...
+    }
+
+    """
+    # importing here due to circular import
+    from edx_sysadmin.constants import (  # pylint: disable=import-outside-toplevel,cyclic-import
```

Comment:
Please move `FIELDS_AND_DEFAULT_VALUES_MAP` here in `utils.py`. Rename `utility.py` to `utils.py'.

Reply:
updated and converted it into a function as it required some functions which should be declared earlier than their usage  and were causing import issue.